fetcher/fetcher.go

```diff
@@ -79,144 +82,147 @@ func (f *Fetcher) runWorker() {
 
 	for {
 		select {
-		case t, ok := <-f.fetchable:
+		case t, ok := <-f.tasks:
 			if !ok {
 				return
 			}
 
-			if ok := f.shouldFetch(t); !ok {
```

Comment:
`shouldFetch` is no longer needed because we only enqueue a key the first time it is requested